server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1047 @@
+
+        :raises: MissingValue if some values are missing
+        :raises: OperationPostponed when an async operation is performed.
+        """
+        body = request.body_as_json
+        missing_params = []
+        units = body.get('units')
+        if units is None:
+            missing_params.append('units')
+        options = body.get('options')
+        if options is None:
+            missing_params.append('options')
+
+        if missing_params:
+            raise MissingValue(missing_params)
+
+        agent_manager = factory.consumer_agent_manager()
+        task = agent_manager.update_content(consumer_id, units, options)
+        raise OperationPostponed(TaskResult.from_task_status_dict(task))
+
+    def uninstall(self, request, consumer_id):
+        """
+        Uninstall content (units) on a consumer.
+
+        Expected body: {units:[], options:<dict>}
+        where unit is: {type_id:<str>, unit_key={}} and the
+        options is a dict of uninstall options.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+
+        :raises: MissingValue if some values are missing
+        :raises: OperationPostponed when an async operation is performed.
+        """
+        body = request.body_as_json
+        missing_params = []
+        units = body.get('units')
+        if units is None:
+            missing_params.append('units')
+        options = body.get('options')
+        if options is None:
+            missing_params.append('options')
+
+        if missing_params:
+            raise MissingValue(missing_params)
+
+        agent_manager = factory.consumer_agent_manager()
+        task = agent_manager.uninstall_content(consumer_id, units, options)
+        raise OperationPostponed(TaskResult.from_task_status_dict(task))
+
+
+class ConsumerHistoryView(View):
+    """
+    View for consumer history retrieval.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id):
+        """
+        Retrieve histroy for the specified consumer
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+
+        :return: Response representing the binding
+        :rtype: django.http.HttpResponse
+        """
+        query_param = request.GET
+        filters = _ensure_input_encoding(query_param)
+        event_type = filters.get('event_type', None)
+        limit = filters.get('limit', None)
+        sort = filters.get('sort', None)
+        start_date = filters.get('start_date', None)
+        end_date = filters.get('end_date', None)
+
+        if sort is None:
+            sort = 'descending'
+
+        if limit:
+            limit = int(limit)
+
+        results = factory.consumer_history_manager().query(consumer_id=consumer_id,
```

Comment:
On 03/25/2015 11:42 AM, Austin wrote:
> This would upset @rbarlow <https://github.com/rbarlow> for using too
> many lines :)

Hehehe

-- 
Randy Barlow